packages/react-components/src/components/resource-explorers/explorers/alarm-explorer/internal-alarm-explorer.tsx

```diff
@@ -0,0 +1,197 @@
+
+import type {
+  AlarmByAssetRequestParameters,
+  AlarmExplorerProps,
+  AlarmResourcesRequestParameters,
+} from './types';
+import {
+  ResourceDropDown,
+  ResourceTable,
+  ResourceExplorerVariant,
+} from '../../variants';
+import {
+  DEFAULT_DEFAULT_PAGE_SIZE,
+  DEFAULT_IS_RESOURCE_DISABLED,
+  DEFAULT_IS_TABLE_FILTER_ENABLED,
+  DEFAULT_IS_TABLE_SEARCH_ENABLED,
+  DEFAULT_IS_TABLE_ENABLED,
+  DEFAULT_IS_TABLE_USER_SETTINGS_ENABLED,
+  DEFAULT_ON_SELECT_RESOURCE,
+  DEFAULT_RESOURCE_EXPLORER_VARIANT,
+  DEFAULT_SELECTED_RESOURCES,
+  DEFAULT_SELECTION_MODE,
+  DEFAULT_SHOULD_PERSIST_USER_CUSTOMIZATION,
+  createDefaultTableUserSettings,
+  DEFAULT_ALARM_RESOURCE_NAME,
+  DEFAULT_PLURAL_ALARM_RESOURCE_NAME,
+  latestValueTimeCellRenderer,
+  latestValueCellRenderer,
+} from '../../constants/defaults';
+import { AlarmResourceWithLatestValue } from '../../types/resources';
+import {
+  DEFAULT_ALARM_TABLE_DEFINITION,
+  createDefaultLatestValuesTableDefinition,
+} from '../../constants/table-resource-definitions';
+import { DEFAULT_ALARM_DROP_DOWN_DEFINITION } from '../../constants/drop-down-resource-definitions';
+import { useUserCustomization } from '../../helpers/use-user-customization';
+import { TableResourceDefinition } from '../../types/table';
+import { useAlarms } from '../../../../hooks/useAlarms';
+import { isAlarmResource, transformAlarmData } from './transformAlarmData';
+import { createNonNullableList } from '../../../../utils/createNonNullableList';
+
+export function InternalAlarmExplorer({
+  iotSiteWiseClient,
+  iotEventsClient,
+  parameters = [],
+  shouldPersistUserCustomization = DEFAULT_SHOULD_PERSIST_USER_CUSTOMIZATION,
+  defaultPageSize = DEFAULT_DEFAULT_PAGE_SIZE,
+  variant = DEFAULT_RESOURCE_EXPLORER_VARIANT,
+  resourceName = DEFAULT_ALARM_RESOURCE_NAME,
+  pluralResourceName = DEFAULT_PLURAL_ALARM_RESOURCE_NAME,
+  isAlarmDisabled = DEFAULT_IS_RESOURCE_DISABLED,
+  selectedAlarms = DEFAULT_SELECTED_RESOURCES,
+  onSelectAlarm = DEFAULT_ON_SELECT_RESOURCE,
+  selectionMode = DEFAULT_SELECTION_MODE,
+  tableResourceDefinition: customTableResourceDefinition,
+  defaultTableUserSettings: customDefaultTableUserSettings,
+  tableSettings: {
+    isTitleEnabled = DEFAULT_IS_TABLE_ENABLED,
+    isSearchEnabled = DEFAULT_IS_TABLE_SEARCH_ENABLED,
+    isFilterEnabled: isTableFilterEnabled = DEFAULT_IS_TABLE_FILTER_ENABLED,
+    isUserSettingsEnabled = DEFAULT_IS_TABLE_USER_SETTINGS_ENABLED,
+  } = {},
+  dropDownResourceDefinition = DEFAULT_ALARM_DROP_DOWN_DEFINITION,
+  dropDownSettings: { isFilterEnabled: isDropDownFilterEnabled = false } = {},
+  ariaLabels,
+  description = '',
+  timeZone,
+  significantDigits,
+}: AlarmExplorerProps) {
+  const requests = useMemo(
+    () =>
+      isAlarmByAssetRequestParameters(parameters)
+        ? parameters.map((p) => ({ assetId: p.assetId }))
+        : parameters.map((p) => ({ assetModelId: p.assetModelId })),
+    [parameters]
+  );
+
+  const alarms = useAlarms({
```

Comment:
We should have the following settings field:
```
settings: {
  fetchOnlyLatest: true,
  refreshRate: Infinity, // Only fetch data once
}
```

Reply:
Yes, good point. We don't have the history yet so I didn't consider that! But the other explorers do refresh every 5s, so we should leave this functionality.